File: answers/C/@mykeels/12-salary-classifier.c
#include <stdio.h>

int main() {
    while (1) {
        printf("Enter a salary amount: ");

        double salary = 0;
        scanf("%lf", &salary);

        if (salary < 0) {
            printf("Invalid Amount");
        }
        else if (salary < 50000) {
            printf("Basic Earner\n");
        }
        else if (salary < 200000) {
            printf("Mid Earner\n");
        }
        else {
            printf("High Earner\n");
        }
    }
    return 0;
}

File: answers/Go/@codehakase/12-salary-classifier.go
package main

import "fmt"

func main() {
	var salary int
	for {
		fmt.Print("Enter salary amount: ")
		fmt.Scanf("%d\n", &salary)
		if salary < 1 {
			fmt.Println("Invalid salary amount")
		}
		if salary < 50000 {
			fmt.Println("Basic Earner")
		} else if salary <= 200000 {
			fmt.Println("Mid Earner")
		}
		if salary > 200000 {
			fmt.Println("High Earner")
		}
	}
}
